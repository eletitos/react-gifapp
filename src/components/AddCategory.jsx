import PropTypes from "prop-types";
import { useState } from "react";

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');
    const onInputChange = (e) =>{
        setInputValue(e.target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length <=1) return;

        onNewCategory(inputValue.trim());
        setInputValue('');
    }

    return (
        <form aria-label="formulario" onSubmit={onSubmit}>
        <input 
            type="text"
            placeholder="Mete aquí tu categoría"
            value={ inputValue }
            onChange= {onInputChange}
        />
        </form>
    );
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}




