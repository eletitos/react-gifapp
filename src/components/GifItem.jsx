export const GifItem = ({title, url}) => {

    return (
        <div className='card'>
            <img src={url} alt={title} loading="lazy" />
            <div>{title}</div>
        </div>
    );
}
