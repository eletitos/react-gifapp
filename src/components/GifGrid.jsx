import useFetchGifs from "../hooks/useFetchGifs";
import GifItem from "./GifItem";


//esto es un componente. 
const GifGrid = ( {category} ) => {   

    const{ images, isLoading } = useFetchGifs( category );
    

    return (
        <>
            <h3> { category  } </h3> 
            <div className="card-grid">
                {images.map((image) =>(
                    <GifItem 
                        key={image.id} 
                        {...image} //Con esto se esparcen las properties
                    />
                ))}
            </div>    
        </>
    );
}

export default GifGrid;
