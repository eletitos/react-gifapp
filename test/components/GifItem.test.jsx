import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Pruebas sobre el componente GifItem', () => {
    const title = 'Titulo de prueba';
    const url = 'https://media.giphy.com/media/hsCpEuXImDcQJ8EsrL/giphy.gif'
        

    
    test('Debe hacer match con el snapshot', () => {

        const {container} = render ( <GifItem title= {title} url={url}/> )

        expect(container).toMatchSnapshot();
    });


    test('debe mostrar imagen con el url y alt indicado', () => {

        render ( <GifItem title= {title} url={url}/> );
        // screen.debug();
        const { src, alt } = screen.getByRole('img');
        
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });
    

    test('Debe de mostrar el tÃ­tulo en el componente', () => {
        render ( <GifItem title= {title} url={url}/> );
        expect(screen.getByText(title)).toBeTruthy();
    });

});
