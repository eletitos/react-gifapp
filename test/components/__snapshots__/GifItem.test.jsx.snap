// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo de prueba"
      loading="lazy"
      src="https://media.giphy.com/media/hsCpEuXImDcQJ8EsrL/giphy.gif"
    />
    <div>
      Titulo de prueba
    </div>
  </div>
</div>
`;
